compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -182,32 +209,70 @@ private class ExtractAPICollector(implicit val ctx: Context) extends ThunkHolder
 
     val selfType = apiType(sym.givenSelfType)
 
-    val name = if (sym.is(ModuleClass)) sym.fullName.sourceModuleName else sym.fullName
+    val name = extractedName(sym)
 
-    val tparams = sym.typeParams.map(apiTypeParameter)
+    val tparams = sym.typeParams.map(apiTypeParameter).toArray
 
     val structure = apiClassStructure(sym)
+    val acc = apiAccess(sym)
+    val modifiers = apiModifiers(sym)
+    val anns = apiAnnotations(sym).toArray
+    val topLevel = sym.isTopLevelClass
+    val childrenOfSealedClass = sym.children.sorted(classFirstSort).map(c =>
+      if (c.isClass)
+        apiType(c.typeRef)
+      else
+        apiType(c.termRef)
+    ).toArray
+
+    val cl = api.ClassLike.of(
+      name, acc, modifiers, anns, defType, api.SafeLazy.strict(selfType), api.SafeLazy.strict(structure), Constants.emptyStringArray,
+      childrenOfSealedClass, topLevel, tparams)
+
+    // if (name.toString.contains("DottyPredef")) {
+    //   println("sym: " + sym)
+    //   println("name: " + name)
+    //   ctx.atPhase(ctx.flattenPhase.next) { implicit ctx =>
+    //     println("flatten: " + sym.fullName.toString)
+    //     println("flattenm: " + sym.fullName.mangledString)
+    //   }
+    //   println("flattenx: " + toDenot(sym.binaryName.toString)
+    // }
+
+    allNonLocalClassesInSrc += cl
+
+    val javaPlatform = ctx.platform.asInstanceOf[JavaPlatform]
```

Comment:
Please avoid the cast here, we don't want the compiler to crash when run with the Scala.js backend. I think we should just move `hasJavaMainMethod` outside of `JavaPlatform`

Reply:
Yes, especially since Scala.js supports exactly the same main methods as Scala/JVM!